src/docfx/lib/UrlUtility.cs

```diff
@@ -344,6 +344,30 @@ public static string RemoveLeadingHostName(string url, string hostName, bool rem
             : $"/{path}";
     }
 
+    public static string GetBookmark(string uid)
+    {
+        var sb = new StringBuilder(uid.ToLowerInvariant());
+        uid = sb
+            .Replace("\"", "")
+            .Replace("'", "")
+            .Replace("%", "")
+            .Replace("^", "")
+            .Replace("\\", "")
+            .Replace("<", "(")
+            .Replace(">", ")")
+            .Replace("[", "(")
+            .Replace("]", ")")
+            .Replace("{", "((")
+            .Replace("}", "))")
+            .ToString();
+        uid = Regex.Replace(uid, @"[^a-zA-Z0-9\(\)\*@]", "-");
+        uid = Regex.Replace(uid, @"^-+", "");
+        uid = Regex.Replace(uid, @"-+$", "");
+        var bookmark = Regex.Replace(uid, @"-+", "-");
+
```

Comment:
would it be more readable to run this line first, then trim start and end *single* `-`

Reply:
Fixed by removing all regex